rotkehlchen/db/addressbook.py

```diff
@@ -90,36 +90,42 @@ def get_addressbook_entries(
 
     def add_addressbook_entries(
             self,
-            book_type: AddressbookType,
+            write_cursor: 'DBCursor',
             entries: list[AddressbookEntry],
+            exists_ok: bool = False,
     ) -> None:
         """
         Add every entry of entries to the address book table.
         If blockchain is None then make sure that the same address doesn't appear in combination
         with other blockchain values.
-        """
-        with self.write_ctx(book_type) as write_cursor:
-            # We iterate here with for loop instead of executemany in order to catch
-            # which identifier is duplicated
-            for entry in entries:
-                try:
-                    # in the case of given blockchain being None delete any other entry for that
-                    # address since they are rendered redundant
-                    if entry.blockchain is None:
-                        write_cursor.execute(
-                            'DELETE FROM address_book where address=? AND blockchain IS NOT NULL',
-                            (entry.address,),
-                        )
 
+        If cursor is given then it is used, otherwise a cursor that corresponds to book_type
+        is generated.
```

Comment:
You resolved my comment, made the cursor a permanent argument but the place where my comment was did you read it? There is a reason I left it on this line.


```suggestion
```